src/main/java/slimeknights/tconstruct/library/tinkering/ToolPart.java

```diff
@@ -0,0 +1,168 @@
+package slimeknights.tconstruct.library.tinkering;
+
+import com.google.common.collect.ImmutableList;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import net.minecraftforge.fml.ModList;
+import net.minecraftforge.fml.loading.moddiscovery.ModInfo;
+import slimeknights.tconstruct.common.config.Config;
+import slimeknights.tconstruct.library.MaterialRegistry;
+import slimeknights.tconstruct.library.Util;
+import slimeknights.tconstruct.library.materials.IMaterial;
+import slimeknights.tconstruct.library.materials.stats.IMaterialStats;
+import slimeknights.tconstruct.library.materials.stats.MaterialStatsId;
+import slimeknights.tconstruct.library.utils.TagUtil;
+import slimeknights.tconstruct.library.utils.Tags;
+import slimeknights.tconstruct.shared.TinkerClient;
+import slimeknights.tconstruct.tools.IToolPart;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+
+public class ToolPart extends MaterialItem implements IToolPart {
+
+  public final MaterialStatsId materialStatId;
+
+  public ToolPart(Properties properties, MaterialStatsId id) {
+    super(properties);
+
+    this.materialStatId = id;
+  }
+
+  @Override
+  public void fillItemGroup(ItemGroup group, NonNullList<ItemStack> items) {
+    if (this.isInGroup(group)) {
+      if (MaterialRegistry.initialized()) {
+        for (IMaterial material : MaterialRegistry.getInstance().getMaterials()) {
+          if (this.canUseMaterial(material)) {
+            items.add(this.getItemstackWithMaterial(material));
+            if (!Config.COMMON.listAllPartMaterials.get()) {
+              break;
+            }
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public boolean canUseMaterial(IMaterial material) {
```

Comment:
I would also put an overridable `canUseMaterial` over in `MaterialItem`. Might let us move the sub items code there, but mostly because that concept is relevant to all material items.